File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit =0, min=prices[0];
        int n = prices.length;
        for(int i=1;i<n;i++){
            int cost = prices[i]-min;
            profit = Math.max(profit,cost);
            min = Math.min(min,prices[i]);
        }
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut max_profit =0;
        let mut min =prices[0];
        
        for &price in prices.iter().skip(1){
            let profit = price - min;

            if profit > max_profit{
                max_profit = profit;
            }
            if price < min{
                min = price;
            }
        }
        max_profit
    }
}